app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.ts

```diff
@@ -229,20 +264,21 @@ export async function switchChain(
 
     await setActiveNetwork(networkClientId);
     res.result = null;
-  } catch (error) {
+  } catch (error: unknown) {
     // We don't want to return an error if user rejects the request
     // and this is a chained switch request after wallet_addEthereumChain.
     // approvalFlowId is only defined when this call is of a
     // wallet_addEthereumChain request so we can use it to determine
     // if we should return an error
     if (
-      error.code === errorCodes.provider.userRejectedRequest &&
+      (error as JsonRpcError).code ===
+        errorCodes.provider.userRejectedRequest &&
```

Comment:
No need for a type assertion here when we can simply validate.

```suggestion
      isJsonRpcError(error)
        && error.code === errorCodes.provider.userRejectedRequest
```